test/inductor/test_cpu_select_algorithm.py

```diff
@@ -1416,6 +1416,85 @@ def forward(self, x, scale):
         vec_amx = VecAMX()
         self._check_amx_counter(vec_amx)
 
+    @inductor_config.patch({"freezing": True})
+    @patches
+    @torch.no_grad
+    # We set allow_ignore_mark_dynamic to True because Dynamo may end up specializing M dimension
+    # despite it being marked as dynamic with mark_dynamic.
+    @dynamo_config.patch({"allow_ignore_mark_dynamic": True})
+    @parametrize("has_bias", [True, False])
+    @parametrize("dtype", [torch.float, torch.bfloat16])
+    @parametrize("per_channel_quant", [True, False])
+    @parametrize("reshape_a", [True, False])
+    @parametrize("dynamic", [True, False])
+    def test_da8w8_sym_act_sym_wgt_with_int_mm(
+        self, has_bias, dtype, per_channel_quant, reshape_a, dynamic
+    ):
+        r"""
+        This testcase check if we can match the int8_dynamic_activation_int8_weight int8 linear pattern from torchao,
+        when activation is symmetrically quantized dynamically & weights are symmetrically quantized (statically)
+        The pattern is:
+            (no bias) _int_mm -> convert_element_type -> ([expand_a] -> mul) -> mul
+        or
+            (with bias) pattern_no_bias -> add
+        Expansion of the scale of activation is optional.
+        The pattern depiction doesn't mean that convert_element_type output is fed into expand_a as input,
+        but simply that activation scale may be applied after an expand operation on it.
+        """
+        if dtype == torch.bfloat16 and not torch.ops.mkldnn._is_mkldnn_bf16_supported():
+            return
+        M = 32
+        in_feature = 48
+        out_feature = 64
+        q_min, q_max = -32, 31
+
+        class Mod(torch.nn.Module):
+            def __init__(self, dtype: torch.dtype, has_bias: bool):
+                super().__init__()
+                self.dtype = dtype
+                self.has_bias = has_bias
+                self.b = torch.randint(
+                    q_min, q_max, [in_feature, out_feature], dtype=torch.int8
+                )
+                self.per_channel_quant = per_channel_quant
+                a_scale_per_tensor = torch.rand([1], dtype=dtype) * 0.01 + 0.01
+                a_scale_per_channel = torch.rand([M, 1], dtype=dtype) * 0.01 + 0.01
+                self.a_scale = (
+                    a_scale_per_channel if per_channel_quant else a_scale_per_tensor
+                )
+                self.b_scale = torch.rand([out_feature]) * 0.01 + 0.01
+                self.b_scale = self.b_scale.to(dtype)
+                self.bias = torch.rand([out_feature], dtype=dtype) if has_bias else None
+
+            def forward(self, a):
+                if reshape_a:
+                    a_reshaped = a.reshape(-1, a.size(-1))
+                else:
+                    a_reshaped = a
+                c = torch._int_mm(a_reshaped, self.b)
+                c = c.to(self.dtype)
+                a_scale = self.a_scale.expand(c.shape)
```

Comment:
Per the description of this test case, it seems this `expand` is optional, but here seems we only tested with expand are present.

Reply:
Thanks, adding corresponding UT helped fix a corner case when an int8 activation is quantized per-tensor, its zero-point is dummy, and its scale is not reshaped (because if a is reshaped, [then it's actually using a different code path - that of x being quantized per-token/per-channel](https://github.com/pytorch/pytorch/blob/e263f0af2373bd1e200532398035b71750f270dd/torch/_inductor/fx_passes/quantization.py#L2713), since the activation scale is applied as a post-op in that case). :

<img width="1913" alt="image" src="https://github.com/user-attachments/assets/901da508-9b4d-4f52-bb8a-70f0ce46e2a4" />

Reply:
> but here seems we only tested with expand are present

We should probably add corresponding tests in `test_mkldnn_pattern_matcher.py` for smooth-quant UTs as well, from where this UT has been adapted.